event/src/unvalidated/event.rs

```diff
@@ -295,10 +295,14 @@ impl TimeEvent {
 /// Raw Time Event as it is encoded in the protocol.
 #[derive(Serialize, Deserialize)]
 pub struct RawTimeEvent {
-    id: Cid,
-    prev: Cid,
-    proof: Cid,
-    path: String,
+    /// The CID of the init event of the stream
+    pub id: Cid,
+    /// The CID of the Data event that is being anchored in the chain
+    pub prev: Cid,
+    /// The CID of the proof block that tells us how to query the chain
+    pub proof: Cid,
+    /// path from the root in the proof block to the prev in the merkle tree
+    pub path: String,
```

Comment:
These should not be public. They are intentionally private so that they cannot be used to accidentally construct nonsense events.

Reply:
RawTimeEvent new is just a pass through and there are getters for every field. The fields are not optional. 
 RawTimeEvent is just a data class I don't see why we would not just have all public fields and no functions on the data class. except for `impl Debug for RawTimeEvent`